packages/@aws-cdk/aws-lambda-python/lib/function.ts

```diff
@@ -34,84 +35,48 @@ export interface PythonFunctionProps extends lambda.FunctionOptions {
   /**
    * The runtime environment. Only runtimes of the Python family are
    * supported.
-   */
-  readonly runtime: lambda.Runtime;
-
-  /**
-   * Determines how asset hash is calculated. Assets will get rebuild and
-   * uploaded only if their hash has changed.
    *
-   * If asset hash is set to `SOURCE` (default), then only changes to the source
-   * directory will cause the asset to rebuild. This means, for example, that in
-   * order to pick up a new dependency version, a change must be made to the
-   * source tree. Ideally, this can be implemented by including a dependency
-   * lockfile in your source tree or using fixed dependencies.
-   *
-   * If the asset hash is set to `OUTPUT`, the hash is calculated after
-   * bundling. This means that any change in the output will cause the asset to
-   * be invalidated and uploaded. Bear in mind that `pip` adds timestamps to
-   * dependencies it installs, which implies that in this mode Python bundles
-   * will _always_ get rebuild and uploaded. Normally this is an anti-pattern
-   * since build
-   *
-   * @default AssetHashType.SOURCE By default, hash is calculated based on the
-   * contents of the source directory. This means that only updates to the
-   * source will cause the asset to rebuild.
+   * @default Runtime.PYTHON_3_7
    */
-  readonly assetHashType?: AssetHashType;
+  readonly runtime?: Runtime;
```

Comment:
Making this optional.

Reply:
Oh I missed that, thanks for the tip! Wouldn't it be nicer to default to Python 3.8 instead to better resolve https://github.com/aws/aws-cdk/issues/10248? That has a much longer EOL, has been around for a few years and is well supported.